src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -248,6 +263,28 @@ public static ValidationException validateKnnField(
             return exception;
         }
 
+        // Return if vector data type does not need to be checked
+        if (expectedVectorDataType == null) {
```

Comment:
I think expectedVectorDataType shouldn't be null?
Also, we should not just return here even if it is null so that the rest of the code can be executed to check other validation; I know there is no other validation now but still it is not future change friendly.

Reply:
When `expectedDimension` is <0 or null, your validation will never get hit.

Reply:
The existing validateKnnField method return one failed validation per time, so I followed same pattern here for vector type.

Reply:
Try to write a unit test. You may find an issue with current implementation.
It is okay to return single exception but the issue here is you are returning when there is no exception. Then the rest of the validation code won't work.
Also, please check if your code works when expectedDimension is < 0 or null. In such case, your validation won't get executed.

Reply:
Offline discussed with Heemin on this part, moved the vector data type check when it's non null before dimension check.